validator/rpc/handlers_keymanager.go

```diff
@@ -238,3 +241,169 @@ func (s *Server) DeleteRemoteKeys(w http.ResponseWriter, r *http.Request) {
 
 	http2.WriteJson(w, RemoteKeysResponse{Data: deleter.DeletePublicKeys(req.Pubkeys)})
 }
+
+// GetGasLimit returns the gas limit measured in gwei defined for the custom mev builder by public key
+func (s *Server) GetGasLimit(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "validator.keymanagerAPI.GetGasLimit")
+	defer span.End()
+
+	if s.validatorService == nil {
+		http2.HandleError(w, "Validator service not ready", http.StatusServiceUnavailable)
+		return
+	}
+
+	rawPubkey := mux.Vars(r)["pubkey"]
+	if rawPubkey == "" {
+		http2.HandleError(w, "pubkey is required in URL params", http.StatusBadRequest)
+		return
+	}
+
+	pubkey, valid := shared.ValidateHex(w, "pubkey", rawPubkey, fieldparams.BLSPubkeyLength)
+	if !valid {
+		return
+	}
+
+	resp := &GetGasLimitResponse{
+		Data: &GasLimitMetaData{
+			Pubkey: rawPubkey,
+		},
+	}
```

Comment:
Could we set the default gas limit first:
```suggestion
	resp := &GetGasLimitResponse{
		Data: &GasLimitMetaData{
			Pubkey: rawPubkey,
			GasLimit: fmt.Sprintf("%d", params.BeaconConfig().DefaultBuilderGasLimit),
		},
	}
```
Then override it if settings are not nil, clear out all the writes and returns and just have them at the very end?
If my explanation is a bit confusing I mean do this:

```
resp := &GetGasLimitResponse{
    Data: &GasLimitMetaData{
	Pubkey:   rawPubkey,
	GasLimit: fmt.Sprintf("%d", params.BeaconConfig().DefaultBuilderGasLimit),
    },
}
settings := s.validatorService.ProposerSettings()
if settings != nil {
	proposerOption, found := settings.ProposeConfig[bytesutil.ToBytes48(pubkey)]
	if found {
		if proposerOption.BuilderConfig != nil {
			resp.Data.GasLimit = fmt.Sprintf("%d", proposerOption.BuilderConfig.GasLimit)
		}
	} else if settings.DefaultConfig != nil && settings.DefaultConfig.BuilderConfig != nil {
		resp.Data.GasLimit = fmt.Sprintf("%d", s.validatorService.ProposerSettings().DefaultConfig.BuilderConfig.GasLimit)
	}
}
http2.WriteJson(w, resp)
```

Reply:
yup we can do that